File: Lecture03/ImplicitIntents/app/src/main/java/com/codingblocks/implicitintents/MainActivity.java
package com.codingblocks.implicitintents;

import android.content.Intent;
import android.net.Uri;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    EditText etUrl;
    Button btnView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etUrl = findViewById(R.id.etUrl);
        btnView = findViewById(R.id.btnClick);

//        String url = etUrl.getText().toString();

        btnView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                String url = "tel:" + etUrl.getText().toString();
//
//                Uri uriToView = Uri.parse(url);
//
//                //tel:9968195588
//
//                Intent i = new Intent();
//
//                i.setAction(Intent.ACTION_DIAL);
//
//                i.setData(uriToView);
//
//                startActivity(i);

                Intent i = new Intent();
                i.setAction("xyz");


//                i.putExtra("NAME", )

                startActivity(i);
            }
        });

    }
}

File: Lecture20/Alarms/app/src/main/java/com/codingblocks/alarms/MainActivity.kt
package com.codingblocks.alarms

import android.app.AlarmManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.SystemClock
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val alarmManager: AlarmManager = getSystemService(Context.ALARM_SERVICE) as AlarmManager

        val intent = Intent(this, MainActivity::class.java)

        val pi = PendingIntent.getActivity(
            this,
            0,
            intent,
            PendingIntent.FLAG_UPDATE_CURRENT
        )

        btnSet.setOnClickListener {

            val triggerTime = etInterval.text.toString().toLong()

            val calendar = Calendar.getInstance().apply {
                timeInMillis = System.currentTimeMillis()
                set(Calendar.MONTH, 12)
                set(Calendar.DAY_OF_MONTH, 25)
                set(Calendar.YEAR, 2019)
                set(Calendar.HOUR_OF_DAY, 20)
                set(Calendar.MINUTE, 30)
            }


            //Triggers an alarm at 8:30 pm on December 25th, 2019
            alarmManager.set(
                AlarmManager.RTC,
                calendar.timeInMillis,
                pi
            )

            alarmManager.setRepeating(
                AlarmManager.ELAPSED_REALTIME,
                SystemClock.elapsedRealtime() + triggerTime * 1000,
                60 * 1000,
                pi
            )
        }

    }
}
